driver/controller.go

```diff
@@ -92,11 +92,13 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		}, nil
 	}
 
+	tierMapper := map[string]string{"maxiops": upcloud.StorageTierMaxIOPS, "hdd": upcloud.StorageTierHDD}
+	tier := tierMapper[req.Parameters["tier"]]
```

Comment:
@mescudi21 , here you didn't check if req.Parameters["tier"] holding a valid value, so if it is not, you silently have tier = "", as it will be added as a new key into the map with empty value.